python/src/iceberg/types.py

```diff
@@ -93,21 +128,37 @@ def precision(self) -> int:
     def scale(self) -> int:
         return self._scale
 
-    def __eq__(self, other):
-        if type(self) is type(other):
-            return self.precision == other.precision and self.scale == other.scale
-        return False
 
+class NestedField(IcebergType):
+    """equivalent of `NestedField` type from Java implementation"""
```

Comment:
Doc strings should use normal sentence case, so this should start with a capital letter.

Also, I don't think that we want to refer to the Java implementation. This represents a field of a struct, a map key, a map value, or a list element. This is where field IDs are held.